analysis.go

```diff
@@ -24,83 +27,22 @@ type Info struct {
-// https://github.com/openshift/origin/blob/master/vendor/k8s.io/kubernetes/pkg/api/types.go
-// However the fields that were not required for our purposes were removed for brevity.
-// Currently the copied definitions are:
-// - ContainerStateWaiting
-
-// Event is a representation of the items in the OpenShift event log, this is
-// trimmed to only the required fields.
-type Event struct {
-	Kind           string `json:"kind"`
-	InvolvedObject struct {
-		Namespace string `json:"namespace"`
-		Name      string `json:"name"`
-	} `json:"involvedObject"`
-	Reason  string `json:"reason"`
-	Message string `json:"message"`
-	Count   int    `json:"count"`
-	Type    string `json:"type"`
-}
-
-// Events is a representation of everything in the OpenShift event log for a
-// particular project.
-type Events struct {
-	Items []Event `json:"items"`
+	CheckName     string        `json:"checkName" yaml:"checkName"`
+	Status        int           `json:"status" yaml:"status"`
+	StatusMessage string        `json:"statusMessage" yaml:"statusMessage"`
+	Info          []Info        `json:"info,omitempty" yaml:"info"`
+	Events        []types.Event `json:"events,omitempty" yaml:"events"`
 }
 
-// DeploymentConfigs is a representation of the OpenShift deployment configs.
-type DeploymentConfigs struct {
-	Items []struct {
-		Kind     string `json:"kind"`
-		Metadata struct {
-			Name      string `json:"name"`
-			Namespace string `json:"namespace"`
-		} `json:"metadata"`
-		Spec struct {
-			Replicas int `json:"replicas"`
-		} `json:"spec"`
-	} `json:"items"`
-}
-
-// ContainerStateWaiting is one possible status that a container can be in.
-type ContainerStateWaiting struct {
-	// A brief CamelCase string indicating details about why the container is in waiting state.
-	Reason string `json:"reason,omitempty"`
-	// A human-readable message indicating details about why the container is in waiting state.
-	Message string `json:"message,omitempty"`
-}
-
-// Pods is a representation of all the pods in a project from OpenShift.
-type Pods struct {
-	Items []struct {
-		Metadata struct {
-			Name      string `json:"name"`
-			Namespace string `json:"namespace"`
-		} `json:"metadata"`
-		Status struct {
-			ContainerStatuses []struct {
-				Name  string `json:"name"`
-				State struct {
-					Waiting *ContainerStateWaiting `json:"waiting,omitempty"`
-				} `json:"state"`
-			} `json:"containerStatuses"`
-		} `json:"status"`
-	} `json:"items"`
+// CheckResults stores the result of a check in a scope, .... FIXME: need to review this, both Scope and Name are being set to a project name.
+type CheckResults struct {
+	Scope   string `json:"scope"`
+	Name    string `json:"name,omitempty"`
+	Results []Result
 }
 
-// CheckTask is the interface which checks must implement.
-type CheckTask func(DumpedJSONResourceFactory) (Result, error)
-
 // AnalysisResults is a representation of the JSON analysis results file.
 type AnalysisResults map[string]map[string][]Result
```

Comment:
The trouble with this being a `map[string]...` is that we lose some of the type guarantees and information. One can have arbitrary string keys here, vs. a struct that gives you a prescriptive and restrictive set of keys in the output.